app/controllers/duties_controller.rb

```diff
@@ -88,6 +92,17 @@ def grabable?(duty_id_params)
     end
   end
 
+  def drop_eligible?(drop_duty_ids)
+    duties = duties_sorted_by_start_time(drop_duty_ids)
+    first_timeslot = duties[0].timeslot_id
```

Comment:
More idiomatic: `first_start_time = duties.first.timeslot.time_range.start_time`

You can chain always entities with relationships

Reply:
And avoid `array[0]’ in favour of `array.first’

Reply:
I see, thanks!